src/KeyConnector/Services/GoogleCloudKmsRsaKeyService.cs

```diff
@@ -26,14 +26,14 @@ public class GoogleCloudKmsRsaKeyService : IRsaKeyService
 
         public async Task<byte[]> EncryptAsync(byte[] data)
         {
-            var result = await _keyManagementServiceClient.EncryptAsync(_cryptoKeyName, ByteString.CopyFrom(data));
-            return result.Ciphertext.ToByteArray();
-
+            var publicKey = await GetRsaPublicKeyAsync();
+            var result = publicKey.Encrypt(data, RSAEncryptionPadding.OaepSHA256);
```

Comment:
Rather than using the `RsaEncryptAsync` method in `CryptoFunctionService`, I'm doing it manually here to change the encryption algorithm. The encryption in `CryptoFunctionService` uses SHA1, and customer success has asked that we update to SHA256 to avoid complaints from customers.

This also allows us to use a `2048 bit RSA - OAEP Padding - SHA256 Digest` key in GCP. If we encrypt with SHA1, we have to use the SHA1 version of the key.

Any reasons we should continue to use SHA1?

Reply:
I know sha1 sounds scary, but using sha1 with Oaep is not a problem as far as I know.

Reply:
I don't think you need to call into crypto function service. We aren't doing that for the Verify call below.